c/zssLogging.c

```diff
@@ -43,6 +45,182 @@ bool isLogLevelValid(int level) {
+      case LOG_COMP_ZLCOMPILER: snprintf(suffix,sizeof(LOG_COMP_TEXT_ZLCOMPILER),LOG_COMP_TEXT_ZLCOMPILER);
+                    break;
+      case LOG_COMP_ZLRUNTIME: snprintf(suffix,sizeof(LOG_COMP_TEXT_ZLRUNTIME),LOG_COMP_TEXT_ZLRUNTIME); 
+                    break;
+      case LOG_COMP_STCBASE: snprintf(suffix,sizeof(LOG_COMP_TEXT_STCBASE),LOG_COMP_TEXT_STCBASE);
+                    break;
+      case LOG_COMP_HTTPSERVER: snprintf(suffix,sizeof(LOG_COMP_TEXT_HTTPSERVER),LOG_COMP_TEXT_HTTPSERVER);
+                    break;
+      case LOG_COMP_DISCOVERY: snprintf(suffix,sizeof(LOG_COMP_TEXT_DISCOVERY),LOG_COMP_TEXT_DISCOVERY);
+                    break;
+      case LOG_COMP_DATASERVICE: snprintf(suffix,sizeof(LOG_COMP_TEXT_DATASERVICE),LOG_COMP_TEXT_DATASERVICE);
+                    break;
+      case LOG_COMP_CMS: snprintf(suffix,sizeof(LOG_COMP_TEXT_CMS),LOG_COMP_TEXT_CMS);
+                    break;
+      case 0xC0001: snprintf(suffix,sizeof(LOG_COMP_TEXT_CMS),LOG_COMP_TEXT_CMS);
+                    break;
+      case 0xC0002: snprintf(suffix,6,"cmspc");
+                    break;
+      case LOG_COMP_LPA: snprintf(suffix,sizeof(LOG_COMP_TEXT_LPA),LOG_COMP_TEXT_LPA);
+                    break;
+      case LOG_COMP_RESTDATASET: snprintf(suffix,sizeof(LOG_COMP_TEXT_RESTDATASET),LOG_COMP_TEXT_RESTDATASET);
+                    break;
+      case LOG_COMP_RESTFILE: snprintf(suffix,sizeof(LOG_COMP_TEXT_RESTFILE),LOG_COMP_TEXT_RESTFILE);
+                    break;
+    }
+  }
+  else if (compID >= LOG_PROD_ZIS && compID < LOG_PROD_ZSS) {
+    snprintf(prefix,sizeof(LOG_PREFIX_ZIS),LOG_PREFIX_ZIS);
+  }
+  else if (compID >= LOG_PROD_ZSS && compID < LOG_PROD_PLUGINS ) {
+    snprintf(prefix,sizeof(LOG_PREFIX_ZSS),LOG_PREFIX_ZSS); 
+    switch (id) {
+      case LOG_COMP_ID_ZSS: snprintf(suffix,sizeof(LOG_COMP_ID_TEXT_ZSS),LOG_COMP_ID_TEXT_ZSS);
+                    break;
+      case LOG_COMP_ID_CTDS: snprintf(suffix,sizeof(LOG_COMP_ID_TEXT_CTDS),LOG_COMP_ID_TEXT_CTDS);
+                    break;
+      case LOG_COMP_ID_SECURITY: snprintf(suffix,sizeof(LOG_COMP_ID_TEXT_SECURITY),LOG_COMP_ID_TEXT_SECURITY);
+                    break;
+      case LOG_COMP_ID_UNIXFILE: snprintf(suffix,sizeof(LOG_COMP_ID_TEXT_UNIXFILE), LOG_COMP_ID_TEXT_UNIXFILE);
+                    break;
+    }
+  }
+  else if (compID > LOG_PROD_PLUGINS) {
+    snprintf(suffix,sizeof(component->name),"%s",(strrchr(component->name, '/')+1)); // given more characters than it writes
+    snprintf(prefix,sizeof(component->name)-sizeof(suffix),"%s",component->name);
+  }
+
+  char *filename;
+  filename = strrchr(path, '/'); // returns a pointer to the last occurence of '/'
+  filename+=1; 
+  //               formatting + prefix + suffix + filename + line number
+  int locationSize =  LOCATION_PREFIX_PADDING  + strlen(prefix) + strlen(suffix) + strlen(filename) + LOCATION_SUFFIX_PADDING; 
+  *locationInfo = (char*) safeMalloc(locationSize,"locationInfo");
```

Comment:
I think it's a really bad idea to allocate storage in a logging function. Even so, where is the NULL check? Where do you free this storage?